src/Engine/Component/SoundComponents/SoundComponent.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#ifndef _SOUND_COMPONENT_HPP_
+#define _SOUND_COMPONENT_HPP_
+
+#include "glm/glm.hpp"
+
+#include "Component/Component.hpp"
+
+#ifdef HAVE_FMOD_LIBRARY
+#include "ThirdParty/fmod/fmod.hpp"
```

Comment:
Is FMOD included as third party libraries like ImGui and rapidJSON? Or is it external like glm?

Reply:
I thought you wanted it to be treated like a third party, so I put all the header files in the Third Party folder. The libraries themselves are external because they are operating system specific, and you guys didn't want them in the repo.

Reply:
I'm confused. Why do we have both external libraries and internal headers? Why not keep all of FMOD either external or internal?

Reply:
You said, "can we not treat this as a third party?", so I took that as "He wants the header's to be in Third Party", so I put the headers in Third Party. That's why they are internal. The libraries were internally in the repo, then you guys said, "Why do you have the libraries in the repo? Those should be external." So I had the libraries external. I am getting a lot of mixed signals, and it's starting to get really frustrating because I'm having to keep changing according to what you guys want, but then are never satisfied with what I do. In order for FMOD to be internal, the library files have to be somewhere in the repo in order to be linked. Since you all are apparently opposed to that, I can have the header files be included in the Cmake file, which was what was the case in the first place over a week ago.

Reply:
It sounds like there was some miscommunication. My understanding was we should keep FMOD completely external and expect everyone to be able to install it and set up the environment variables themselves. 
At this point let's just do whatever is easiest to move forward. With this PR is FMOD somewhere in between internal and external? Headers are included in ThirdParty, but team members are still required to install FMOD and set up the environmental variables?